instrumentation/jetty-httpclient/jetty-httpclient-9.2/library/src/main/java/io/opentelemetry/javaagent/instrumentation/jetty/httpclient/v9_2/JettyHttpClient9TracingInterceptor.java

```diff
@@ -0,0 +1,168 @@
+        Response.CompleteListener {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(JettyHttpClient9TracingInterceptor.class);
+
+  @Nullable private Context context;
+
+  @Nullable
+  public Context getContext() {
+    return this.context;
+  }
+
+  private final Context parentContext;
+
+  public JettyHttpClient9TracingInterceptor(Context parentCtx) {
+    this.parentContext = parentCtx;
+  }
+
+  public void attachToRequest(Request jettyRequest) {
+    List<JettyHttpClient9TracingInterceptor> current =
+        jettyRequest.getRequestListeners(JettyHttpClient9TracingInterceptor.class);
+
+    if (!current.isEmpty()) {
+      LOG.warn("A tracing interceptor is already in place for this request! ");
+      return;
+    }
+    startSpan(jettyRequest);
+
+    // wrap all important request-based listeners that may already be attached, null should ensure
+    // all listeners
+    // are returned here
+    List<Request.RequestListener> existingListeners = jettyRequest.getRequestListeners(null);
+    wrapRequestListeners(existingListeners);
+
+    jettyRequest
+        .onRequestBegin(this)
+        .onRequestFailure(this)
+        .onResponseFailure(this)
+        .onResponseSuccess(this);
+  }
+
+  private void wrapRequestListeners(List<Request.RequestListener> requestListeners) {
+
+    List<Request.BeginListener> beginListeners =
+        requestListeners.stream()
+            .filter(rl -> rl instanceof Request.BeginListener)
+            .map(rl -> (Request.BeginListener) rl)
+            .collect(Collectors.toList());
+    List<Request.FailureListener> failureListeners =
+        requestListeners.stream()
+            .filter(rl -> rl instanceof Request.FailureListener)
+            .map(rl -> (Request.FailureListener) rl)
+            .collect(Collectors.toList());
+
+    //    Context context = (this.ctx != null) ? this.ctx : this.parentContext;
+    Context context = this.parentContext;
+    for (Request.BeginListener blOriginal : beginListeners) {
+      requestListeners.set(
+          requestListeners.indexOf(blOriginal),
+          (Request.BeginListener)
+              request -> {
+                try (Scope ignore = context.makeCurrent()) {
+                  blOriginal.onBegin(request);
+                }
+              });
+    }
+    for (Request.FailureListener flOriginal : failureListeners) {
+      requestListeners.set(
+          requestListeners.indexOf(flOriginal),
+          (Request.FailureListener)
+              (request, throwable) -> {
+                try (Scope ignore = context.makeCurrent()) {
+                  flOriginal.onFailure(request, throwable);
+                }
+              });
+    }
+  }
+
+  private void startSpan(Request request) {
+
+    if (this.parentContext != null) {
+      if (!tracer().shouldStartSpan(this.parentContext)) {
+        return;
+      }
+      Context context = tracer().startSpan(this.parentContext, request, request);
+      this.context = context;
```

Comment:
Nit: there's no need to assign context to a local variable, you can set the field directly:
```suggestion
      this.context = tracer().startSpan(this.parentContext, request, request);
```

Reply:
Done!  Thanks @mateuszrzeszutek